### DISCUSSION TITLE
Announcing v3.0.0 Beta 1

### DISCUSSION THREAD

**Discussion by nikkow** [nikkow/node-red-contrib-tahoma#53] [Announcements]
Hello,

As many of you noticed, and reported in the issues section, this plugin used to rely on cloud-based APIs that were not properly working. 

The version 1.x relied on the undocumented Tahoma API. Over time, Somfy added constraints to this API that led to the endpoints being throttled and unusable by third-party softwares. How can I blame them for that? The usage was, after all, not really legitimate 😅 .

In 2020, Somfy came up with what appeared to be the solution to all these issues: the Open API. This plugin was completely rewritten (using newer technologies, like TypeScript) to use this brand-new API. However, the promises were not fully met. Some of us encountered many issues whilst creating the application on the developer platform, or several session expired based scenarios. After many discussions and emails exchanged with the Somfy development team, no solution could be found...  

A few months ago, a new announcement from Somfy came online: the local API. Yes, another API... Would it finally be the solution to solve our issues? Will it be worth investing some time to adapt the module? I needed to run some tests first, to ensure that this solution would work in the long term. To my biggest surprise: it did. This new API only relies on the cloud to generate an identifier (token) that will be directly sent to your Tahoma box. This basically means that your node-red instance will directly talk to the box itself, not to the cloud. And this token seems to live forever.

**Today, I released the first beta version of v3.0.0 implementing this new API.** 🎉 

It is available for you to test through the beta channel. The instructions can be found [here](https://github.com/nikkow/node-red-contrib-tahoma/releases/tag/v3.0.0-beta.1).

If you encounter any issues, please remember this is still in **beta** and do not hesitate to raise an issue 😉 

Thank you all!
N. ✌️

**Comment by whoami-pi** [comment]
Thanks for all your effort. Do you know if this local API works also with the connexxoon Box?

**Reply by nikkow** [reply]
Hi @whoami-pi,

I cannot confirm this, but according to this issue: https://github.com/Somfy-Developer/Somfy-TaHoma-Developer-Mode/issues/9, Connexoon is not yet supported. Somfy hasn't communicated any details on if and when it'll be supported.

**Reply by svefro** [reply]
Somfy says to me that it will not be supported due to a hardware limitation.

**Comment by svefro** [comment]
Somfy actually shipped a tahoma box to me for free since the api is no longer supported by the connexion box. (in norway we have 5 year waranty on consumer electronics)

I have also been communicating with the developer team and complained about the "open" api witch did not support my awning.
it ended up as one way communication where i sent them an email every time the api was down and got no response. (long email thread :p)

 I had made my own implementation of it in node red but now i want to switch to node-red-contrib-tahoma. i have activated developer mode and generated a token with 3.0.0-beta.5. but none of my devices show up in the device list. is this a known issue?
i get a gateway url and then press "update". nothing in node red output and no devices in the list.

After restarting node red the dropdown box is disabled but the config node still shows the gateway url
